pkg/core/native/native_neo.go

```diff
@@ -1045,20 +1077,26 @@ func (n *NEO) getAccountState(ic *interop.Context, args []stackitem.Item) stacki
 // ComputeNextBlockValidators returns an actual list of current validators.
 func (n *NEO) ComputeNextBlockValidators(blockHeight uint32, d *dao.Simple) (keys.PublicKeys, error) {
 	numOfCNs := n.cfg.GetNumOfCNs(blockHeight + 1)
-	// Most of the time it should be OK with RO cache, thus try to retrieve
-	// validators without RW cache creation to avoid cached values copying.
-	cache := d.GetROCache(n.ID).(*NeoCache)
```

Comment:
Option 2: _always_ take the value from cache and _never_ compute here. Update the list in PostPersist if needed (if it's `nil` effectively), it'll be pushed into the lower DAO in a standard way then. This adds a bit of additional load to non-CN nodes, but at the same time doesn't change much for CN ones.

Reply:
I like this approach, it's similar to the standard way of working with simple contract values, implemented.